spec/util/extensions/miq-benchmark_spec.rb

```diff
@@ -58,4 +63,26 @@
     end
     expect(Benchmark.in_realtime_block?).to be_falsey
   end
+
+  it "Timeout raising within .realtime_block" do
+    expect(Benchmark.in_realtime_block?).to be_falsey
+    50_000.times do |i|
+      timeout = 10**Random.rand(-13.0..-10.0)
```

Comment:
Wait this is funny, 1e-10 is already 0.1ns, less than a cpu cycle!  (Moore proof haha).
My "distribution" showed that timeouts in this range do trigger the bug, sometimes, but I think it just expired immediately and depended on thread context switch...

I'll measure with longer values.